File: backend2/src/main/java/com/expense/tracker/model/Expense.java
package com.expense.tracker.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "expenses")
@Data
@NoArgsConstructor
public class Expense {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @Column(name = "item_name", nullable = false)
    private String itemName;

    private Double quantity;

    private String unit; // kg, litre, pieces, etc.

    @Column(name = "price_per_unit")
    private BigDecimal pricePerUnit;

    @Column(name = "total_amount", nullable = false)
    private BigDecimal totalAmount;

    @Column(nullable = false)
    private LocalDate date;

    private String notes;

    @Column(name = "bill_image_url")
    private String billImageUrl;
    
    @Column(name = "is_recurring")
    private boolean isRecurring;

    @Column(name = "recurring_frequency")
    private String recurringFrequency; // DAILY, WEEKLY, MONTHLY
}

File: backend2/models/expense.go
package models

import (
	"time"
)

type Expense struct {
	ID                 uint      `gorm:"primaryKey"`
	UserID             uint      `gorm:"not null"`
	CategoryID         *uint     
	ItemName           string    `gorm:"column:item_name;not null"`
	Quantity           *float64
	Unit               string    // kg, litre, pieces, etc.
	PricePerUnit       *float64  `gorm:"column:price_per_unit;type:decimal(19,2)"`
	TotalAmount        float64   `gorm:"column:total_amount;not null;type:decimal(19,2)"`
	Date               time.Time `gorm:"not null;type:date"`
	Notes              string
	BillImageURL       string    `gorm:"column:bill_image_url"`
	IsRecurring        bool      `gorm:"column:is_recurring"`
	RecurringFrequency string    `gorm:"column:recurring_frequency"` // DAILY, WEEKLY, MONTHLY
	
	// Associations
	User     User     `gorm:"foreignKey:UserID"`
	Category *Category `gorm:"foreignKey:CategoryID"`
}
